dato/js/recview.js

```diff
@@ -0,0 +1,158 @@
+$(document).ready( function(event){
+  // Loads Predictive Service client
+  // once it loads it triggers the recommenderView function
+  function loadScript(url){
+
+      var script = document.createElement("script")
+      script.type = "text/javascript";
+
+      if (script.readyState){  //IE
+          script.onreadystatechange = function(){
+              if (script.readyState == "loaded" ||
+                      script.readyState == "complete"){
+                  script.onreadystatechange = null;
+                  callback();
+              }
+          };
+      } else {  //Others
+          script.onload = function(){
+              recommenderView();
+          };
+      }
+
+      script.src = url;
+      document.getElementsByTagName("head")[0].appendChild(script);
+  }
+
+  loadScript('http://static.dato.com/files/scripts/predictive-service-client/ps-client.js');
+
+  // Snowplow function to get id from cookie
+  function getSnowplowDuid(cookieName) {
+    cookieName = cookieName || '_sp_';
+    var matcher = new RegExp(cookieName + 'id\\.[a-f0-9]+=([^;]+);');
+    var match = document.cookie.match(matcher);
+    if (match && match[1]) {
+      return match[1].split('.')[0];
+    } else {
+      return false;
+    }
+  }
+
+  function recommenderView() {
+
+    // function to construct element tree for recommender view
+    // if resp.data.response array is empty it skips creating recommender section
+    function renderView(err, resp) {
+      if(err) {
+        console.log(err);
+        return;
+      }
+
+      if(resp.data.response.length){
+        const section = document.createElement('section');
+        section.id = 'recview';
+        const h2 = document.createElement('h2');
+        h2.innerText = 'Recommended Resources';
+        const root = document.createElement('div');
+        root.id = 'root';
+        const list = document.createElement('ul');
+        list.id = 'list';
+
+        root.appendChild(list);
+        section.appendChild(h2);
+        section.appendChild(root);
+        $('#section-').after(section);
+
+        function feedback(event){
+          event.preventDefault();
+          var uuid = resp.data.uuid;
+          var itemId = event.target.href;
+          onClick(uuid, itemId);
+        }
+        for (let i in resp.data.response) {
+          let listEl = document.createElement('li');
+          let link = document.createElement('a');
+          link.className = 'recommendation';
+          link.href = resp.data.response[i].page_url;
+          link.innerText = resp.data.response[i].title;
+          link.addEventListener('click', feedback);
+          listEl.appendChild(link);
+
+          document.getElementById('list').appendChild(listEl);
+        }
+
+        // cb event for mouse over event
+        // tracks when user first hovers over recommender section
+        // sends feedback with data
+        function mouseOverCB(event) {
+          console.log(resp.data.uuid);
```

Comment:
We should remove console logging used for debugging before going to production (this will have a minor perf hit, and users will see it in the console).